pkg/master/master.go

```diff
@@ -707,9 +712,10 @@ func (m *Master) getExtensionResources(c *Config) map[string]rest.Storage {
 		deploymentStorage := deploymentetcd.NewStorage(restOptions("deployments"))
 		storage["deployments"] = deploymentStorage.Deployment
 		storage["deployments/status"] = deploymentStorage.Status
-		// TODO(madhusudancs): Install scale when Scale group issues are fixed (see issue #18528).
-		// storage["deployments/scale"] = deploymentStorage.Scale
 		storage["deployments/rollback"] = deploymentStorage.Rollback
+		if !m.ApiGroupVersionOverrides["autoscaling/v1"].Disable {
```

Comment:
registered != turned on. You're checking here if it's in runtime config, which would be used for turning hpa on and off. I agree it won't work if it's not in api versions-- but that's not what you're checking. @nikhiljindal to confirm